### PR TITLE
Add clear method and test

### LINKED ISSUE #6: Synchronize repo with most up-to-date code from ingest / metabolize
The ingest codebase has code for the Bulk class that has deviated from this repo -- we need to merge in the changes located in ingest so the two are in-sync (ingest is more up to date and complete).

### PR DESCRIPTION
**Pull request by nwarlen**
Required to fix [Ingest Issue #12](https://github.com/CascadeEnergy/sensei3-measurement-digest-ingest/issues/12). 

Also fixes #6

### CODE REVIEW

File: src/Bulk.php [hunk 1]
```diff
@@ -174,4 +174,13 @@ public function flush()
...
     }
+
+    /**
+     * Clears all items in the Bulk object.
+     */
+    public function clear()
+    {
+        $this->itemList = [];
+        $this->itemCount = 0;
+    }
```
Comment (cberube) [inline src/Bulk.php]: Now that this function exists, it should really be called in places where the code currently explicitly clears the item list. (e.g. line 174 & 175 above).

File: tests/BulkTest.php [hunk 2]
```diff
@@ -135,4 +135,12 @@ public function testItShouldRaiseAnExceptionIfTheErrorListIsNotEmpty()
         $this->bulk->addItem('idFoo', 'foo');
         $this->bulk->flush();
     }
+
+    public function testItShouldEmptyTheItemListOnClear()
+    {
+        $this->bulk->addItem('idfoo','foo');
+        $this->assertEquals($this->bulk->getItemCount(), 1);
+        $this->bulk->clear();
+        $this->assertEquals($this->bulk->getItemCount(), 0);
```
Comment (cberube) [inline tests/BulkTest.php]: You can assert that the item list is empty as well with `assertAttributeEquals` (I think that's the method name -- something close to that at least).

**Comment by jinsyaoommen** [comment]
:ok:

**Comment by cberube** [comment]
:ok:

**Pull request merged**